src/modules/service-catalog/components/useItemFormFields.tsx

```diff
@@ -27,16 +27,24 @@ const formatField = (field: TicketField): Field => {
   };
 };
 
-const isAssociatedLookupField = (field: TicketField) => {
+const isAssociatedLookupField = (
+  field: TicketField,
+  setAssociatedLookupField: (field: Field | null) => void
+) => {
   const customObjectKey = getCustomObjectKey(
     field.relationship_target_type as string
   );
-  return customObjectKey === "service_catalog_item";
+  if (customObjectKey === "service_catalog_item") {
+    setAssociatedLookupField(formatField(field));
+    return true;
+  }
+  return false;
 };
 
 const fetchTicketFields = async (
   form_id: string,
-  baseLocale: string
+  baseLocale: string,
+  setAssociatedLookupField: (field: Field | null) => void
```

Comment:
I think that passing a setter callback to a function called `fetchTicketFields` and then again to a function called `isAssociatedLookupField` is a bit convoluted and not really clean, since these functions are more getter. Probably it would be better to return both form fields and the lookup field from this function and call the setter outside

Reply:
Right, I changed that